### PR TITLE
added BogoSort in sorting algorithms and changed mistakes in heap sort

### LINKED ISSUE #167: [Feature Request] BogoSort in Sorting algorithms
### Describe the solution you'd like
add bogo sort in sorting algorithms

### Additional context
---

**Would you like to work on this feature?**  
- [x] Yes  
screencapture-localhost-3000-algo-docs-algorithms-sorting-algorithms-bogo-sort-algo-2024-10-07-16_59_51

- [ ] No

### PR DESCRIPTION
**Pull request by Anandha-Vihari**
### Description
added BogoSort in sorting algorithms,explained bogosort and changed a minor mistake in heapsort
screencapture-localhost-3000-algo-docs-algorithms-sorting-algorithms-bogo-sort-algo-2024-10-07-16_59_51

Fixes #167 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/sorting-algorithms/BogoSort.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    cout << endl;
+
+    return 0;
+}
+```
+
+### Summary:
+
+Bogo sort is a highly inefficient sorting algorithm based on generating random permutations until the array is sorted. With an average-case time complexity of O((n+1)!), it is impractical for anything but small arrays or as a teaching tool to illustrate inefficiency. Despite its impracticality, Bogo sort serves as a humorous example of how not to sort a list, given its extreme inefficiency.
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/BogoSort.md]: replace `O((n+1)!)` with `$O((n+1)!)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: docs/algorithms/sorting-algorithms/BogoSort.md (hunk 1 above)
Comment (Anandha-Vihari) [reply docs/algorithms/sorting-algorithms/BogoSort.md]: @ajay-dhangar yeah i replaced O((n+1)!) to $O((n+1)!)$

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by RadhikaMalpani1702** [comment]
@mastansayyad please review this PR

**Pull request merged**